src/analyze/analyze-dump.c

```diff
@@ -92,6 +117,17 @@ static int dump_patterns(sd_bus *bus, char **patterns) {
         _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL, *m = NULL;
         int r;
 
+        if (strv_contains(patterns, "memory")) {
+                strv_remove(patterns, "memory");
+
+                r = dump_memory(bus);
+                if (r < 0)
+                        return r;
+
+                if (strv_isempty(patterns))
+                        return 1; /* We are done */
+        }
```

Comment:
just make this a verb on its own, and make it take the service name to issue this on as argument, after all this is entirely generic and can be implemented in any service. (and we probably should)

Reply:
done, ptal